src/frontend/komponenter/Behandling/Brev/Brevmeny.tsx

```diff
@@ -51,39 +51,50 @@ interface Props {
     settKanSendesTilBeslutter: (kanSendesTilBeslutter: boolean) => void;
 }
 
-const Brevmeny: React.FC<Props> = ({
-    settBrevRessurs,
-    behandlingId,
-    personopplysninger,
-    settKanSendesTilBeslutter,
-}) => {
+const brevMal = 'innvilgetOvergangsstonadHoved2';
+const datasett = 'ef-brev';
+
+const initFlettefelterMedVerdi = (brevStruktur: BrevStruktur): FlettefeltMedVerdi[] =>
+    brevStruktur.flettefelter.flettefeltReferanse.map((felt) => ({
+        _ref: felt._id,
+        verdi: null,
+    }));
+
+const Brevmeny: React.FC<Props> = (props) => {
     const { axiosRequest } = useApp();
     const [brevStruktur, settBrevStruktur] = useState<Ressurs<BrevStruktur>>(byggTomRessurs());
-    const [alleFlettefelter, settAlleFlettefelter] = useState<FlettefeltMedVerdi[]>([]);
-    const [valgteFelt, settValgteFelt] = useState<ValgtFelt>({});
-    const [valgteDelmaler, settValgteDelmaler] = useState<{ [delmalNavn: string]: boolean }>({});
-
-    const brevMal = 'innvilgetOvergangsstonadHoved2';
-    const datasett = 'ef-brev';
 
     useEffect(() => {
         axiosRequest<BrevStruktur, null>({
             method: 'GET',
             url: `/familie-brev/api/${datasett}/avansert-dokument/bokmaal/${brevMal}/felter`,
         }).then((respons: Ressurs<BrevStruktur>) => {
-            if (respons.status === RessursStatus.SUKSESS) {
-                settAlleFlettefelter(
-                    respons.data.flettefelter.flettefeltReferanse.map((felt) => ({
-                        _ref: felt._id,
-                        verdi: null,
-                    }))
-                );
-                settBrevStruktur(respons);
-            }
+            settBrevStruktur(respons);
         });
         // eslint-disable-next-line
     }, []);
 
+    return (
+        <DataViewer response={{ brevStruktur }}>
+            {({ brevStruktur }) => <BrevmenyView {...props} brevStruktur={brevStruktur} />}
+        </DataViewer>
+    );
+};
+
+const BrevmenyView: React.FC<Props & { brevStruktur: BrevStruktur }> = ({
+    settBrevRessurs,
+    behandlingId,
+    personopplysninger,
+    settKanSendesTilBeslutter,
+    brevStruktur,
+}) => {
+    const { axiosRequest } = useApp();
+    const [alleFlettefelter, settAlleFlettefelter] = useState<FlettefeltMedVerdi[]>(
+        initFlettefelterMedVerdi(brevStruktur)
```

Comment:
Er dette egentligen riktig måte? 
Eller burde det være 

```
const [alleFlettefelter, settAlleFlettefelter] = useState<FlettefeltMedVerdi[]>([]);
    useEffect(() => {
        settAlleFlettefelter(initFlettefelterMedVerdi(brevStruktur))
    }, [brevStruktur]);
```

Reply:
hade anvænd useEffect om `alleFlettefelter` var avhænig av props

Reply:
Når endrer brevstruktur seg? Kun ved lasting av data? Isåfall trenger du ikke en useEffect, men bare behold det som initiell state